src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java

```diff
@@ -200,17 +200,22 @@ public void invalidateUserJwt(final Cookie customJWTCookie, final HttpServletRes
         response.addCookie(thirdPartyAuthToken);
     }
 
-    public void validateRoles(List<Role> userRoles) {
+    public void validateRoles(List<Role> userRoles, String payloadRoles) {
         final Set<String> formattedUserRoles = userRoles.stream()
                 .map(role -> roleMapper.roleToRoleDto(role).getName())
                 .collect(Collectors.toSet());
-        if(formattedUserRoles.isEmpty()){
-            throw new UnauthorizedException("User is blocked");
+        boolean userHasBeenUnblocked = payloadRoles.equals("[]") && formattedUserRoles.size() > 0;
+
+        if (formattedUserRoles.isEmpty()) {
+            throw new UnauthorizedException("Payload is invalid - User is blocked");
+        }
+        if (userHasBeenUnblocked) {
+            throw new UnauthorizedException("Payload is invalid - User has been unblocked");
```

Comment:
This bit seems v backwards. "You're unblocked so therefore you're not authenticated", trying to think of a better solution

Reply:
As discussed a cleaner way might be to upon hitting the login endpoint and having a valid JWT, update the JWT with roles from the DB

Reply:
Was more of a "Your payload is out of date therefore you're not authenticated". Eventually we will have more general role validation here (rather than just checking block vs unblock)